### ISSUE TITLE
Incorporate support for multiple workflows into FACTS.py

### ISSUE THREAD

**Issue by bobkopp** [radical-collaboration/facts#87]
Core to the application of FACTS are workflows with overlapping sets of modules -- eg workflow 1 might include modules A, B and C, while workflow 2 might include modules A, B and D. In the IPCC application, unnecessary duplication was avoided by manually copying output files into workflow directories before running the totaling scripts on each directly. Would be nice to have a way of specifying this within the experiment config.yml file so one all SLR modules used could be run in one experiment step, then the totaling and ESL steps could be applied in parallel to the appropriate collections of outputs. 

Perhaps an "include_in_workflow" option that leads to link creation and "apply_to_workflow" option that points the totaling and ESL scripts in the right direction would work?

**Comment by bobkopp** [comment]
@mturilli and @andre-merzky, what's the cleanest way to handle this in EnTK?

**Comment by bobkopp** [comment]
Option: add a include_in_workflow option to the experiment config.yml that can take multiple values. As pipelines are initiated with this option, a dictionary accumulates listing the location of the output files in $SHARED that go with each work flow. After all pipelines are completed, this dictionary is written to a .yml file in $SHARED. Modify the totaling module so this can be taken as input in lieu of a directory pointer and, if provided, have that module loop over all workflows defined in the yml.

**Comment by bobkopp** [comment]
Started this in the workflows branch. runFACTS.py now creates a workflows.yml file in the experiment directory. Still need the mechanics for the total module to use this file.

**Comment by andre-merzky** [comment]
`$SHARED` is always local to a single run and thus to the sandbox of that run. EnTK though is also able to reference staging data from outside the session sandbox, i.e. from a previous run.  In your example, one option would be to run modules `A` and `B` stand alone, and when running the workflows which contain `C` or `D`, to reference the output data from `A` and `B` from their original sandbox.

Would that approach address the problem, or is the request explicitly to run both workflows within the same session?

**Comment by bobkopp** [comment]
I think I have a plan here, and have started implementing. But the key is to be able to run multiple workflows in one sandbox, so that we don’t duplicate module runs unnecessarily.

The approach is to create a workflows.yml file in the experiment step while all the sea level modules are running, then have the totaling step generate a pipeline for totaling each workflow defined in this file. It seems to be working in tests from the command line; there will be a challenge in ensuring that all file locations are correctly identified once we are running in EnTK.
________________________________

**Comment by andre-merzky** [comment]
We can do run multiple workflows in the same sandbox.  I will have to confirm that data staging across those workflows works as expected though.  Right now, `FACTS.py` returns a set of pipelines, i.e., a single workflow.  I assume that the approach you plan is to return a set of such workflows, i.e., a set of a set of pipelines?  Once that happens, I'll change `runFACTS.py` to execute those workflows one after the other.

**Comment by andre-merzky** [comment]
Oh, I see.  Well, then in that case (single set of pipelines) it will be considered a single large workflow on the EnTK level, and we should see no issues with data staging etc.

**Comment by bobkopp** [comment]
Ok, this should be addressed by https://github.com/radical-collaboration/facts/pull/89 (at least, it produces reasonable looking runFACTS.py --debug output).
